TradeItIosTicketSDK2/TradeItOrdersTableViewManager.swift

```diff
@@ -0,0 +1,240 @@
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        return self.orderSectionPresenters[section].header(forTableView: tableView)
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return self.orderSectionPresenters[section].heightForHeaderInSection()
+    }
+    
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return CGFloat.leastNormalMagnitude
+    }
+    
+    // MARK: Private
+    
+    private func addRefreshControl(toTableView tableView: UITableView) {
+        let refreshControl = UIRefreshControl()
+        refreshControl.attributedTitle = NSAttributedString(string: "Refreshing...")
+        refreshControl.addTarget(
+            self,
+            action: #selector(initiateRefresh),
+            for: UIControlEvents.valueChanged
+        )
+        TradeItThemeConfigurator.configure(view: refreshControl)
+        tableView.addSubview(refreshControl)
+        self.refreshControl = refreshControl
+    }
+    
+    /**
+     * This is to split orders in order to have a specific section for group orders
+    **/
+    private func getSplittedOrdersArray(orders: [TradeItOrderStatusDetails]) -> [[TradeItOrderStatusDetails]]{
+        return orders.reduce([[]], { splittedArrays, order in
+            var splittedArraysTmp = splittedArrays
+            let lastResult: [TradeItOrderStatusDetails] = splittedArrays[(splittedArrays.endIndex - 1)]
+            
+            if !order.isGroupOrder() && !lastResult.contains(order) { // this is not a group order, we can append the order
+                splittedArraysTmp[(splittedArraysTmp.endIndex - 1)].append(order)
+                return splittedArraysTmp
+            } else { // This is a group order or the begining of a new array
+                splittedArraysTmp.append([order])
+                return splittedArraysTmp
+            }
+        })
+    }
+    
+    private func buildOrderSectionPresentersFrom(splitedOrdersArray: [[TradeItOrderStatusDetails]]) {
+        splitedOrdersArray.forEach { splittedOrders in
+            var orders = splittedOrders
+            var title = ""
+            var isGroupOrder = false
+            if let groupOrder = (splittedOrders.filter { $0.isGroupOrder()}).first
+                , let groupOrderType = groupOrder.groupOrderType {
+                title = groupOrderType.lowercased().replacingOccurrences(of: "_", with: " ").capitalizingFirstLetter()
+                orders = splittedOrders.flatMap { $0.groupOrders ?? [] }
+                isGroupOrder = true
+            }
+            self.orderSectionPresenters.append(
+                OrderSectionPresenter(
+                    orders: orders,
+                    title: title,
+                    isGroupOrder: isGroupOrder
+                )
+            )
+        }
+    }
+
```

Comment:
This function is a little bit hard to follow - `forEach` usually indicates that there's is a functional way to do this. Maybe think about moving some of those conditions and `title` decisions to a separate function. Using `map` is probably going to clean this up as well.